tsconfig.json で strict オプションを有効にするとコンパイルエラーになる

typescriptの `strict` オプションを有効にした状態でビルドすると以下のようにエラーになりました。

```
node_modules/@akashic-extension/akashic-hover-plugin/lib/HoverPlugin.d.ts:11:5 - error TS2416: Property 'operationTrigger' in type 'HoverPlugin' is not assignable to the same property in base type 'OperationPlugin'.
  Type 'Trigger<OperationPluginOperation>' is not assignable to type 'Trigger<(string | number)[] | OperationPluginOperation>'.
    Types of property '_handlers' are incompatible.
      Type 'TriggerHandler<OperationPluginOperation>[]' is not assignable to type 'TriggerHandler<(string | number)[] | OperationPluginOperation>[]'.
        Type 'TriggerHandler<OperationPluginOperation>' is not assignable to type 'TriggerHandler<(string | number)[] | OperationPluginOperation>'.
          Type '(string | number)[] | OperationPluginOperation' is not assignable to type 'OperationPluginOperation'.
            Type '(string | number)[]' is not assignable to type 'OperationPluginOperation'.

11     operationTrigger: g.Trigger<g.OperationPluginOperation>;
       ~~~~~~~~~~~~~~~~

  node_modules/@akashic/akashic-engine/lib/main.d.ts:5486:9
    5486         data: (number | string)[];
                 ~~~~
    'data' is declared here.


Found 1 error.

```

利用したライブラリのバージョン等は[エラー再現用プロジェクト](https://github.com/pocketberserker/akashic-hover-plugin-200-compile-error-test) を参考にしてください（Node.js は 10.x 系です）。

--- a/.gitignore
+++ b/.gitignore
@@ -1,5 +1,8 @@
-node_modules
 .DS_Store*
-npm-debug.log
-lib
-tmp
+*.log
+*.iml
+*.tgz
+
+node_modules/
+lib/
+doc/

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,4 +1,7 @@
 # CHANGELOG
- 
+
+## 2.0.1
+* TypeScript の `strict: true` 指定時にビルドが失敗する件の修正
+
 ## 2.0.0
 * 初期リリース

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@akashic-extension/akashic-hover-plugin",
-  "version": "2.0.0",
+  "version": "2.0.1",
   "description": "A plugin handling mouse hover/unhover events easily for Akashic Engine",
   "main": "lib/index.js",
   "typings": "lib/index.d.ts",
@@ -19,15 +19,15 @@
     "README.md"
   ],
   "dependencies": {
-    "@akashic/akashic-engine": "~2.4.11"
+    "@akashic/akashic-engine": "~2.4.14"
   },
   "devDependencies": {
-    "@types/node": "7.0.5",
+    "@types/node": "12.0.4",
     "remark-cli": "6.0.1",
     "remark-lint": "~6.0.4",
     "rimraf": "2.6.3",
-    "tslint": "~4.4.2",
-    "typescript": "~3.3.0"
+    "tslint": "~5.17.0",
+    "typescript": "~3.5.1"
   },
   "repository": {
     "type": "git",

--- a/sample/package.json
+++ b/sample/package.json
@@ -13,7 +13,7 @@
   "author": "",
   "license": "",
   "dependencies": {
-    "@akashic-extension/akashic-hover-plugin": "~2.0.0"
+    "@akashic-extension/akashic-hover-plugin": "~2.0.1"
   },
   "devDependencies": {
     "@akashic/akashic-cli-scan": "~0.2.2",

--- a/sample/src/main.ts
+++ b/sample/src/main.ts
@@ -1,8 +1,8 @@
 import * as hover from "@akashic-extension/akashic-hover-plugin";
-import { HoverableRect } from "./HoverableRect";
 import { HoverableLabel } from "./HoverableLabel";
+import { HoverableRect } from "./HoverableRect";
 
-export function main(param: g.GameMainParameterObject) {
+export function main(param: g.GameMainParameterObject): void {
 	const scene = new g.Scene({game: g.game, assetIds: ["aco"]});
 	scene.loaded.addOnce(() => {
 		const font = new g.DynamicFont({
@@ -15,7 +15,7 @@ export function main(param: g.GameMainParameterObject) {
 			(() => {
 				const aco = new g.FrameSprite({
 					scene: scene,
-					src: <g.ImageAsset>scene.assets["aco"],
+					src: scene.assets["aco"] as g.ImageAsset,
 					x: 5,
 					y: i * 50,
 					width: 32,

--- a/sample/tsconfig.json
+++ b/sample/tsconfig.json
@@ -4,7 +4,8 @@
     "lib": ["es5", "dom"],
     "noImplicitAny": true,
     "module": "commonjs" ,
-    "outDir": "./script"
+    "outDir": "./script",
+    "strict": true
   },
   "files": [
     "node_modules/@akashic/akashic-engine/lib/main.d.ts"

--- a/src/HoverPlugin.ts
+++ b/src/HoverPlugin.ts
@@ -11,12 +11,12 @@ export class HoverPlugin implements g.OperationPlugin {
 
 	game: g.Game;
 	view: HTMLElement;
-	beforeHover: HoverableE;
-	operationTrigger: g.Trigger<g.OperationPluginOperation>;
+	beforeHover: HoverableE | null;
+	operationTrigger: g.Trigger<g.OperationPluginOperation | (number | string)[]>;
 
-	_onMouseMove_bound: () => void;
-	_onMouseOut_bound: () => void;
-	_getScale: () => {x: number, y: number};
+	_onMouseMove_bound: (e: MouseEvent) => void;
+	_onMouseOut_bound: (e: MouseEvent) => void;
+	_getScale: (() => {x: number, y: number}) | null;
 
 	_cursor: string;
 	_showTooltip: boolean;
@@ -25,7 +25,7 @@ export class HoverPlugin implements g.OperationPlugin {
 		this.game = game;
 		this.view = (view as any).view;
 		this.beforeHover = null;
-		this.operationTrigger = new g.Trigger<g.OperationPluginOperation>();
+		this.operationTrigger = new g.Trigger();
 		this._cursor = option.cursor || "pointer";
 		this._showTooltip = !!option.showTooltip;
 		this._getScale = (view as any).getScale ? () => (view as any).getScale() : null;
@@ -83,7 +83,7 @@ export class HoverPlugin implements g.OperationPlugin {
 
 	_onUnhovered(target: HoverableE): void {
 		this.view.style.cursor = "auto";
-		if (this.beforeHover.unhovered) {
+		if (this.beforeHover && this.beforeHover.unhovered) {
 			this.beforeHover.unhovered.fire();
 			if (this._showTooltip) {
 				this.view.removeAttribute("title");

--- a/src/HoverPlugin.ts
+++ b/src/HoverPlugin.ts
@@ -1,5 +1,5 @@
-import { HoverPluginOptions } from "./HoverPluginOptions";
 import { HoverableE } from "./HoverableE";
+import { HoverPluginOptions } from "./HoverPluginOptions";
 
 /**
  * ホバー機能を提供するプラグイン。

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -5,6 +5,7 @@
     "noImplicitAny": true,
     "module": "commonjs" ,
     "outDir": "./lib",
+    "strict": true,
     "declaration": true
   },
   "files": [

--- a/tslint.json
+++ b/tslint.json
@@ -9,7 +9,7 @@
       "parameters",
       "statements"
     ],
-    "array-type": [false],
+    "array-type": [true, "array"],
     "ban": false,
     "class-name": true,
     "comment-format": [true,
@@ -22,9 +22,22 @@
     "interface-name": [false],
     "jsdoc-format": true,
     "label-position": true,
-    "max-classes-per-file": [true, 1],
-    "max-line-length": [true, 200],
+    "max-classes-per-file": [false, 1],
+    "max-line-length": [false, 140],
     "member-access": [false],
+    "member-ordering": [true, {
+      "order": [
+        "public-static-field",
+        "private-static-field",
+        "public-instance-field",
+        "private-instance-field",
+        "public-constructor",
+        "private-constructor",
+        "public-instance-method",
+        "protected-instance-method",
+        "private-instance-method"
+      ]
+    }],
     "object-literal-shorthand": false,
     "prefer-for-of": false,
     "trailing-comma": [true, {
@@ -46,11 +59,11 @@
     "no-empty": true,
     "no-empty-interface": false,
     "no-eval": true,
+    "no-parameter-properties": true,
     "no-string-literal": false,
     "no-switch-case-fall-through": true,
     "no-trailing-whitespace": true,
     "no-unused-expression": true,
-    "no-unused-variable": [true],
     "no-use-before-declare": false,
     "no-var-requires": true,
     "object-literal-sort-keys": false,
@@ -60,13 +73,15 @@
       "check-else",
       "check-whitespace"
     ],
-    "ordered-imports": [false],
+    "only-arrow-functions": false,
+    "ordered-imports": [true],
     "arrow-parens": false,
     "quotemark": [true, "double"],
     "radix": true,
     "semicolon": [true, "always"],
     "triple-equals": [true, "allow-null-check"],
     "typedef": [true,
+      "call-signature",
       "property-declaration",
       "member-variable-declaration",
       "parameter"

